File: app/src/lib/schemas/contextCard.ts
import { z } from 'zod';

export const TimeRangeSchema = z.object({
  start: z.string(), // Format: "MM:SS" or minutes as string
  end: z.string(),   // Format: "MM:SS" or minutes as string
});

export const ContextCardSchema = z.object({
  id: z.string().optional(),
  meeting_id: z.string().optional(),
  segment_index: z.number().int().min(0),
  topic: z.string().optional(),
  key_points: z.array(z.string()).default([]),
  decisions: z.array(z.string()).default([]),
  action_items: z.array(z.string()).default([]),
  speakers: z.array(z.string()).default([]),
  time_range: TimeRangeSchema,
  generated_at: z.string(), // ISO timestamp
});

export type ContextCard = z.infer<typeof ContextCardSchema>;
export type TimeRange = z.infer<typeof TimeRangeSchema>;

// For frontend usage without optional fields
export const CreateContextCardSchema = ContextCardSchema.omit({ 
  id: true, 
  meeting_id: true 
});

export type CreateContextCard = z.infer<typeof CreateContextCardSchema>;

File: backend/schemas/contextCard.py
"""
Shared Context Card Schema for backend validation
Mirrors the frontend Zod schema structure
"""

from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime

class TimeRange(BaseModel):
    start: str
    end: str

class ContextCard(BaseModel):
    id: Optional[str] = None
    meeting_id: Optional[str] = None
    segment_index: int
    topic: Optional[str] = None
    key_points: List[str] = []
    decisions: List[str] = []
    action_items: List[str] = []
    speakers: List[str] = []
    time_range: TimeRange
    generated_at: str  # ISO timestamp

class CreateContextCard(BaseModel):
    segment_index: int
    topic: Optional[str] = None
    key_points: List[str] = []
    decisions: List[str] = []
    action_items: List[str] = []
    speakers: List[str] = []
    time_range: TimeRange
